doc/contributing/docs/examples.rst

```diff
@@ -1,72 +1,112 @@
 
-===========================================================
-               Examples and templates
-===========================================================
+================================================================================
+Examples and templates
+================================================================================
 
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-               Module and function
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+In this document we explain general guidelines for describing Tarantool API and
+give some examples and templates.
 
-Here is an example of documenting a module (``my_fiber``) and a function
-(``my_fiber.create``).
+Use this checklist for documenting a function or a method:
 
-.. module:: my_fiber
+* Base description
+* Parameters
+* What this function returns (if nothing, write 'none')
+* Return type (if exists)
+* Possible errors (if exist)
+* Complexity factors (if exist)
+* Note re storage engine (if exists)
+* Example(s)
+* Extra information (if needed)
 
-.. function:: create(function [, function-arguments])
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Documenting function
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-    Create and start a ``my_fiber`` object. The object is created and begins to
-    run immediately.
+We describe functions of Tarantool modules via Sphinx directives ``.. module::``
+and ``.. function::``:
 
-    :param function: the function to be associated with the ``my_fiber`` object
-    :param function-arguments: what will be passed to function
+..  code-block:: rst
 
-    :return: created ``my_fiber`` object
-    :rtype: userdata
+    ..  module:: fiber
 
-    **Example:**
+    ..  function:: create(function [, function-arguments])
 
-    .. code-block:: tarantoolsession
+        Create and start a fiber. The fiber is created and begins to run immediately.
 
-        tarantool> my_fiber = require('my_fiber')
-        ---
-        ...
-        tarantool> function function_name()
-                 >   my_fiber.sleep(1000)
-                 > end
-        ---
-        ...
-        tarantool> my_fiber_object = my_fiber.create(function_name)
-        ---
-        ...
+        :param function: the function to be associated with the fiber
+        :param function-arguments: what will be passed to function
 
```

Comment:
And the resulting HTML looks like this:
...

Hint: extract the code sample into a separate file and include it twice.